pkg/render/guardian.go

```diff
@@ -312,6 +316,45 @@ func (c *GuardianComponent) clusterRoleBinding() *rbacv1.ClusterRoleBinding {
 	}
 }
 
+func (c *GuardianComponent) Role() *rbacv1.Role {
```

Comment:
When I added the RBAC permissions for Guardian, I consolidated the ClusterRoles but accidentally missed adding the kube-controller role, which contains the secret RBAC needed to create secrets and delete es users (cleanupDecommissionedElasticsearchUsers).

Reply:
@caseydavenport I think the kube-controllers  is creating some es-gateway secrets:
https://github.com/tigera/calico-private/blob/7cb660419a1175d8591c7adc65e543ea58009f88/kube-controllers/pkg/controllers/elasticsearchconfiguration/reconcile_elasticsearch_configuration.go#L243

...and deleting some Elasticsearch user secret here:
https://github.com/tigera/calico-private/blob/7cb660419a1175d8591c7adc65e543ea58009f88/kube-controllers/pkg/controllers/elasticsearchconfiguration/reconcile_elasticsearch_configuration.go#L333